### PR TITLE
add number plugin

### PR DESCRIPTION
**Pull request by hbroer**
Adding Number Plugin to format decimal numbers.

### CODE REVIEW

File: src/lib/currency-plugin.ts [hunk 1]
```diff
@@ -4,27 +4,41 @@ import {Parser} from "./plugin-parser";
...
+
+            let num: number;
+            if (pattern === undefined) {
+                num = 0;
+            } else if (
+                options.replace &&
+                options.replace[pattern] !== undefined &&
+                typeof options.replace[pattern] !== "object"
+            ) {
+                num = parseFloat(options.replace[pattern].toString());
```
Comment (maltemattheis) [inline src/lib/currency-plugin.ts]: could be `NaN` right here resulting in a `"NaN"` text. 
But `[[currency|five]]` will result in `0 $`. To be consistent you should add an `|| 0` as you do two lines below.

File: src/lib/number-plugin.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+            let num: number;
+            if (pattern === undefined) {
+                num = 0;
+            } else if (
+                options.replace &&
+                options.replace[pattern] !== undefined &&
+                typeof options.replace[pattern] !== "object"
+            ) {
+                num = parseFloat(options.replace[pattern].toString());
```
Comment (maltemattheis) [inline src/lib/number-plugin.ts]: could be `NaN` right here resulting in a `"NaN"` text. 
But `[[number|five]]` will result in `0`. To be consistent you should add an `|| 0` as you do two lines below.

File: src/lib/currency-plugin.ts (hunk 1 above)
Comment (hbroer) [reply src/lib/currency-plugin.ts]: Correct. Thank you, you are right :)

File: src/lib/number-plugin.ts (hunk 2 above)
Comment (hbroer) [reply src/lib/number-plugin.ts]: And here too. :)

**Pull request merged**